homeassistant/components/meteo_france/sensor.py

```diff
@@ -1,168 +1,231 @@
+        return SENSOR_TYPES[self._type][ENTITY_UNIT]
 
     @property
     def icon(self):
         """Return the icon."""
-        return SENSOR_TYPES[self._type][SENSOR_TYPE_ICON]
+        return SENSOR_TYPES[self._type][ENTITY_ICON]
 
     @property
     def device_class(self):
-        """Return the device class of the sensor."""
-        return SENSOR_TYPES[self._type][SENSOR_TYPE_CLASS]
-
-    def update(self):
-        """Fetch new state data for the sensor."""
-        try:
-            self._client.update()
-            self._data = self._client.get_data()
-
-            if self._type == "weather_alert":
-                if self._alert_watcher is not None:
-                    self._alert_watcher.update_department_status()
-                    self._state = self._alert_watcher.department_color
-                    _LOGGER.debug(
-                        "weather alert watcher for %s updated. Proxy have the status: %s",
-                        self._data["name"],
-                        self._alert_watcher.proxy.status,
-                    )
-                else:
-                    _LOGGER.warning(
-                        "No weather alert data for location %s", self._data["name"]
-                    )
-            else:
-                self._state = self._data[self._type]
-        except KeyError:
-            _LOGGER.error(
-                "No condition %s for location %s", self._type, self._data["name"]
-            )
-            self._state = None
+        """Return the device class."""
+        return SENSOR_TYPES[self._type][ENTITY_CLASS]
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return SENSOR_TYPES[self._type][ENTITY_ENABLE]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {ATTR_ATTRIBUTION: ATTRIBUTION}
+
+    @property
+    def available(self):
+        """Return if state is available."""
+        return self.coordinator.last_update_success
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    async def async_update(self):
+        """Only used by the generic entity update service."""
+        if not self.enabled:
+            return
+
+        await self.coordinator.async_request_refresh()
+
+    async def async_added_to_hass(self):
+        """Subscribe to updates."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+
+class MeteoFranceRainSensor(MeteoFranceSensor):
+    """Representation of a Meteo-France rain sensor."""
+
+    @property
+    def state(self):
+        """Return the state."""
+        next_rain_date_locale = self.coordinator.data.next_rain_date_locale()
+        return (
+            dt_util.as_local(next_rain_date_locale) if next_rain_date_locale else None
```

Comment:
The timestamp device class sensor entity should return isoformatted utc absolute time, not local time.